File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet();
        int maxSubsequenceLength = 0;

        for (int num: nums){
            set.add(num);
        } 

        for (int num: nums){
            int counter = 1; 
            while (set.contains(num - 1)){
                counter += 1;
                num = num - 1;
            }
            if (counter > maxSubsequenceLength){
                maxSubsequenceLength = counter;
            }
        }

        return maxSubsequenceLength;
        
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()


        while True:
            n_sum = 0

            for num in str(n):
                n_sum += int(num) * int(num)

            if n_sum == 1:
                return True
            elif n_sum in seen:
                return False
            else:
                seen.add(n_sum)

            n = n_sum
        